community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -240,16 +245,16 @@ public void failure()
     @Override
     public boolean shouldBeTerminated()
     {
-        return terminated;
+        return termination.shouldBeTerminated();
```

Comment:
I'm probably just being silly, but this will now yield a reference on the heap to be traversed every time we check this flag, which every operation on the Kernel API always do. Would it be a bad idea to have KTI implement TxTermination directly?